libs/api/metadata-converter/src/lib/gn4/gn4.field.mapper.ts

```diff
@@ -288,10 +289,12 @@ export class Gn4FieldMapper {
       ),
     resourceType: (output, source) => {
       const resourceType = getFirstValue(selectField(source, 'resourceType'))
-      const kind = resourceType === 'service' ? 'service' : 'dataset'
+      const kind = getResourceType(resourceType)
+      const reuseType = getReuseType(resourceType)
       return {
         ...output,
         kind,
+        reuseType,
```

Comment:
```suggestion
        ...(reuseType || { reuseType }),
```
We should not write this key if the reuseType is `undefined`

Reply:
Right! I updated the gn4 field mapper and added a ut for this specific use case.